lib/app.js

```diff
@@ -113,4 +123,61 @@ App.bootstrap = function (app, root) {
   });
 };
 
+/**
+ * Performs a graceful shutdown of `app` with the standard
+ * "close" and "free" interceptor flow(s).
+ * @memberof App.prototype
+ * @param {function} cb Root directory of the application
+ * @returns {undefined} No return value.
+ * @api public
+ */
+App.prototype.close = function close(cb) {
+  var app = this;
+  var closeConfig = app.config.get('close') || {};
+  function forceKill() {
+    debug('close timeout');
+    cb(new Error('close timeout'));
+  }
+  var freeAfter = typeof closeConfig.freeAfter === 'number' ?
+    closeConfig.freeAfter :
+    0;
+  // TCP default timeout is 180s
+  var killAfter = typeof closeConfig.killAfter === 'number' ?
+    closeConfig.killAfter :
+    Infinity;
+  debug('shutdown starting, free after %d s and force kill after %d s',
+    freeAfter,
+    killAfter);
+  if (killAfter !== Infinity) {
+    setTimeout(forceKill, killAfter * 1000).unref();
+  }
+  app.perform('close', function superclose(next) {
```

Comment:
does this override the `perform('close')` broadway does?

Reply:
lets go with "kind of", refactoring to remove the collision